bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/PanelContainerItemImpl.java

```diff
@@ -32,7 +32,11 @@ public PanelContainerItemImpl(@NotNull SlingHttpServletRequest request, @NotNull
         ValueMap valueMap = resource.adaptTo(ValueMap.class);
         if (valueMap != null) {
             String jcrTitle = valueMap.get(JcrConstants.JCR_TITLE, String.class);
-            title = valueMap.get(PN_PANEL_TITLE, jcrTitle);
+            if (jcrTitle == null) {
```

Comment:
I think a simple `title = valueMap.get(PN_PANEL_TITLE, valueMap.get(PN_PANEL_TITLE, String))` should suffice here and is way easier to read

Reply:
That would be the same as before, just in one line, and with the same problem, `ValueMap.get(@NotNull String name, @NotNull Class<T> type)` is `@Nullable`, but `defaultValue` in `ValueMap.get(@NotNull String name, @NotNull T defaultValue)` is `@NotNull`.

Reply:
Right, but I think in this case just returning `""` instead of `null` does not do any harm (HTL treats both values the same), so I would suggest simplifying to
```
public PanelContainerItemImpl(@NotNull SlingHttpServletRequest request, @NotNull Resource resource) {
        super(request, resource);
        title = resource.getValueMap().get(PN_PANEL_TITLE, resource.getValueMap().get(JcrConstants.JCR_TITLE, ""));
    }
```
For a longer discussion around those nulllness annotations look at https://jira.apache.org/jira/browse/SLING-6609.

Reply:
I agree, returning empty string would be preferable. I updated the pull request.